core/node/api_server/src/web3/namespaces/debug.rs

```diff
@@ -99,53 +186,48 @@ impl DebugNamespace {
             .get_traces_for_l2_block(block_number)
             .await
             .map_err(DalError::generalize)?;
-        let call_trace = call_traces
-            .into_iter()
-            .map(|call_trace| {
-                let result = Self::map_call(call_trace, only_top_call);
-                ResultDebugCall { result }
-            })
-            .collect();
-        Ok(call_trace)
-    }
 
-    pub async fn debug_trace_block_flat_impl(
-        &self,
-        block_id: BlockId,
-        options: Option<TracerConfig>,
-    ) -> Result<Vec<DebugCallFlat>, Web3Error> {
-        let call_trace = self.debug_trace_block_impl(block_id, options).await?;
-        let call_trace_flat = flatten_debug_calls(call_trace);
-        Ok(call_trace_flat)
+        let mut calls = vec![];
+        let mut flat_calls = vec![];
```

Comment:
This looks awkward. IMO, it'd make sense to "inline" `map_call` code taking the tracer type into account:

```rust
match options.tracer {
    SupportedTracers::FlatCallTracer => {
        let mut flat_calls = vec![];
        for (call, tx_hash, tx_index) in call_traces {
            let mut traces = vec![index];
            Self::map_flatten_call(
                call,
                &mut flat_calls,
                &mut vec![tx_index],
                only_top_call,
                tx_index,
                tx_hash,
            );
        }
        Ok(CallTracerBlockResult::FlatCallTrace(flat_calls))
    }
    // ...
}
```